packages/react-query/src/QueryClientProvider.tsx

```diff
@@ -3,46 +3,22 @@ import * as React from 'react'
 import type { QueryClient } from '@tanstack/query-core'
 import type { ContextOptions } from './types'
 
-declare global {
-  interface Window {
-    ReactQueryClientContext?: React.Context<QueryClient | undefined>
-  }
-}
-
 export const defaultContext = React.createContext<QueryClient | undefined>(
   undefined,
 )
-const QueryClientSharingContext = React.createContext<boolean>(false)
 
-// If we are given a context, we will use it.
-// Otherwise, if contextSharing is on, we share the first and at least one
-// instance of the context across the window
-// to ensure that if React Query is used across
-// different bundles or microfrontends they will
-// all use the same **instance** of context, regardless
-// of module scoping.
 function getQueryClientContext(
   context: React.Context<QueryClient | undefined> | undefined,
```

Comment:
Would it make sense to make the type this instead?
```suggestion
  context: React.Context<QueryClient> | undefined,
```

Reply:
I think this will need to stay as it is due to the `defaultContext` type definition + default value above.

Reply:
I'm just thinking, if someone passed their custom context down, what's the case where it's `undefined`?

Reply:
The `defaultContext` has a default value of `undefined` hence the `React.Context<QueryClient | undefined>` type and the context argument for `getQueryClientContext` is optional, hence the second undefined. I could remove the `undefined` as you said if I were to make the default value for the `defaultContext` be something else but I imagine its done this way for a reason that I'm not familiar with.

Reply:
keep in mind that we'll also replace the feature of passing down a custom context with passing down a custom queryClient:

- https://github.com/TanStack/query/issues/4682

so I think it probably doesn't matter :)